From a4e14a47cefae76a8545f1b342e6d95c4958341a Mon Sep 17 00:00:00 2001
Message-Id: <a4e14a47cefae76a8545f1b342e6d95c4958341a.1476965418.git.keller@fs-net.de>
In-Reply-To: <1bbe4f4c985819c8f8ead54288ca1cc8a4b7f51f.1476965418.git.keller@fs-net.de>
References: <1bbe4f4c985819c8f8ead54288ca1cc8a4b7f51f.1476965418.git.keller@fs-net.de>
From: Hartmut Keller <keller@fs-net.de>
Date: Wed, 19 Oct 2016 16:04:20 +0200
Subject: [PATCH 09/12] Remove compiler warnings in src/gal2d-renderer.c

Add casts to avoid compiler warnings about signed/unsigned comparisons
and pointer arguments of different signedness.
---
 src/gal2d-renderer.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/gal2d-renderer.c b/src/gal2d-renderer.c
index 2ecc8ac..bf12c6e 100644
--- a/src/gal2d-renderer.c
+++ b/src/gal2d-renderer.c
@@ -312,7 +312,7 @@ make_current(struct gal2d_renderer *gr, gcoSURF surface)
 	gcmONERROR(gcoSURF_Lock(surface, &physical[0], (gctPOINTER *)&va));    
     gco2D_SetGenericTarget(gr->gcoEngine2d, 
                         &physical[0], 1, 
-                        &stride, 1, 
+                        (gctUINT32 *)&stride, 1, 
                         gcvLINEAR, format,
                         gcvSURF_0_DEGREE, width, height);
 	
@@ -691,7 +691,7 @@ repaint_region(struct weston_view *ev, struct weston_output *output, struct gal2
 		dstrect.right = bb_rects[0].x2;
 		dstrect.bottom = bb_rects[0].y2;
 
-		if(dstrect.right < 0 || dstrect.bottom < 0 || dstrect.left > dstWidth || dstrect.top > dstHeight)
+		if(dstrect.right < 0 || dstrect.bottom < 0 || dstrect.left > (gctINT32)dstWidth || dstrect.top > (gctINT32)dstHeight)
 		{
 			break;
 		}
@@ -794,10 +794,10 @@ repaint_region(struct weston_view *ev, struct weston_output *output, struct gal2
 				}
 				else
 				{
-					dstrect.right = dstrect.right < dstWidth ? dstrect.right : dstWidth;
-					dstrect.bottom = dstrect.bottom < dstHeight ? dstrect.bottom : dstHeight;
-					srcRect.right = srcRect.right < dstWidth ? srcRect.right : dstWidth;
-					srcRect.bottom = srcRect.bottom < dstHeight ? srcRect.bottom : dstHeight;
+					dstrect.right = dstrect.right < (gctINT)dstWidth ? dstrect.right : (gctINT)dstWidth;
+					dstrect.bottom = dstrect.bottom < (gctINT)dstHeight ? dstrect.bottom : (gctINT)dstHeight;
+					srcRect.right = srcRect.right < (gctINT)dstWidth ? srcRect.right : (gctINT)dstWidth;
+					srcRect.bottom = srcRect.bottom < (gctINT)dstHeight ? srcRect.bottom : (gctINT)dstHeight;
 
 					gcmVERIFY_OK(gco2D_CalcStretchFactor(gr->gcoEngine2d, srcRect.right - srcRect.left,
 						dstrect.right - dstrect.left, &horFactor));
-- 
1.7.4.4

